modules/imgproc/src/resize.cpp

```diff
@@ -1796,8 +1800,73 @@ struct HResizeLinearVecU8_X4
     }
 };
 
+#ifdef __ARM_FEATURE_SVE
+// Accuracy test at: ./bin/opencv_test_imgproc --gtest_filter="Imgproc_Resize*"
+struct HResizeLinearVec_16u32f_SVE
+{
+    int operator()(const ushort** src, float** dst, int count, const int* xofs,
+		   const float* alpha, int, int, int cn, int, int xmax ) const
+    {
+	svbool_t allLanes = svptrue_b32();
+	const int nlanes = svcntw();
+	const int len0 = xmax & -nlanes;
+	int dx=0, k =0;
+	for(; k <= count-2; k+=2 )
+	    {
+		const ushort *S0 = src[k];
+		const ushort *S1 = src[k+1];
+		float *D0 = dst[k];
+		float *D1 = dst[k+1];
+		
+		for(dx = 0;dx < len0; dx += nlanes)
```

Comment:
This code reproduces the algorithm written with HAL for the `operator()` of `HResizeLinearVec_X4` at line https://github.com/opencv/opencv/blob/master/modules/imgproc/src/resize.cpp#L1583

The code is written with SVE intrinsics, in a way that can be transferred directly [*] to HAL code: no fancy SVE features are used, other than runtime increments.

[*] Please notice that the SVE code uses vector conversion of integer to float types, whistle the original code uses the scalar conversion followed by vector constructions. The "build integer vector first and then convert to float" might be also used in the original code without breaking HAL compatibility, because HAL has vector conversions of integers to float.

Reply:
Integer arithmetic should be preferred in algorithms if it is possible to avoid accuracy lost due to FP processing. All bit-exact processing is integer-based (`EXACT` / `BE` modes).

Some implementations use "working type" (WT) to define where such conversion happens (or not).

Reply:
> Integer arithmetic should be preferred in algorithms if it is possible to avoid accuracy lost due to FP processing. All bit-exact processing is integer-based (`EXACT` / `BE` modes).
> 

I don't think there is any issue in loss of precision here? The original code does:

1. convert four `u16` into `s32`, one by one
2. convert the four `s32` into `f32`
3. build a vector of four lanes of `f32`
4. do some math on the `f32x4` vectors 

What we do in the SVE version is:
1. build a vector of `u16`
2. widen such vector to a vector of `s32`
3. convert the `s32` vector into an `f32` vector
4. Do some math on the `f32` vector.

The operations at 1.2.3. are the same in both cases, and the ordering of the FP operation on the `f32` vectors in 4. is the same for both cases (unless I am missing something).

My assessment here is that there is no accuracy loss? Or have I missed something?

Reply:
Main point about integer/floating point computations that we should not change them between SIMD backends (keep existing processing).

Reply:
Oh - I see. For the sake of writing a 100% compatible version, we would need to add LUT constructors to the SIMD types so that we can do the scalar `u16`->`f32` conversion before creating the FP vector.

I *could* do that, but given that this code is not supposed to be merged into the main branch, I don't see a point in doing it?